src/ucs/arch/x86_64/cpu.c

```diff
@@ -586,4 +591,67 @@ ucs_status_t ucs_arch_get_cache_size(size_t *cache_sizes)
     return cache_count == UCS_CPU_CACHE_LAST ? UCS_OK : UCS_ERR_UNSUPPORTED;
 }
 
+void *ucs_x86_memcpy_sse_movntdqa(void *dest, const void *source, size_t len)
+{
+#if defined (__SSE4_1__)
+    const char *src = source;
+    char *dst = dest;
+    void *result = dst;
```

Comment:
i think we should return void from ucs_memcpy_nontemporal, on all archs
this way would not need to save 'result'

Reply:
This was done to make its behavior identical to memcpy which returns dst. The return value is not really useful for anything.

Reply:
let's remove, we don't have to be compliant with glibc's memcpy